browser-interface/packages/shared/world/runtime-7/engine.ts

```diff
@@ -47,7 +61,30 @@ function getUserData(userId: string) {
   }
 }
 
+function getScene(id: string | number): SceneWorker | undefined {
+  if (typeof id === 'string') return getSceneWorkerBySceneID(id)
+  if (typeof id === 'number') return getSceneWorkerBySceneNumber(id)
+  return undefined
+}
+
+function getEngineFromScene(scene?: SceneWorker): IEngine | undefined {
+  return scene?.rpcContext.internalEngine?.engine
+}
+
 export const localProfileChanged = mitt<LocalProfileChange>()
+export const audioStreamEmitter = mitt<AudioStreamChange>()
+
+// AudioStream updates
+audioStreamEmitter.on('changeState', ({ entityId, sceneId, state }) => {
+  const scene = getScene(sceneId)
+  const engine = getEngineFromScene(scene)
+
+  if (!engine) return
+
+  const AudioEvent = defineAudioEvent(engine)
+  AudioEvent.addValue(entityId, { state, timestamp: Date.now() })
```

Comment:
```suggestion
const audioEvent = AudioEvent.get(entityId)
const lastAudio = [...audioEvent.values()].pop()
const timestamp = (lastAudio.timestamp ?? 0) + 1
AudioEvent.addValue(entityId, { state, timestamp })
```

Reply:
mmmm this will not be the actual timestmap in which the state changed 🤔

Reply:
I think the tiemstamp is a monotonic counter for ordering the values (like the CRDT lamport timestamp) , not the actual timestamp in which the event happens.